### ISSUE TITLE
“get_user” only works for models, not pornstars

### ISSUE THREAD

**Issue by github-ph-52obr** [Egsagon/PHUB#27]
The `get_user` function doesn’t seem to work with pornstars, only for models. E.g. if the user’s url is in the www.pornhub.com/pornstar/[pornstar-name] format, then the get_user function doesn’t work. If the user’s url is in the www.pornhub.com/model/[model-name] format then it works.

Code used when passing a pornstar name:

```
user = client.get_user('this-user')
for video in user.videos:
    print(video.title)
```

**Comment by Egsagon** [comment]
This was probably due to some update on the website. Should be fixed now, thanks for reporting.

**Comment by github-ph-52obr** [comment]
How do I download this latest release?

Using `pip install --upgrade git+https://github.com/Egsagon/PHUB.git` tells me that I already have the latest installed:

![IMG_3865](https://github.com/Egsagon/PHUB/assets/106784630/76d0952c-6161-4991-a008-ea563247c73e)

I’m using this in a docker container, which makes it harder for me to just copy over the query.py file. So I need to update from the terminal.

**Comment by Egsagon** [comment]
I forgot to change the version number. It should work fine now.

**Comment by github-ph-52obr** [comment]
It’s still telling me it installed v.4.1.3.post1 and not v4.1.4 that has the fix.

**Comment by Egsagon** [comment]
Try specifying a specific version:
`pip install git+github.com/egsagon/phub.git@v4.1.4`

**Comment by Egsagon** [comment]
Try specifying a specific version:
`pip install git+github.com/egsagon/phub.git@v4.1.4`

**Comment by github-ph-52obr** [comment]
Strange, that reverts it back to v4.1.3:

![IMG_3894](https://github.com/Egsagon/PHUB/assets/106784630/4ff542db-d9f9-48d8-a8dc-f1f6f01f6969)

**Comment by Egsagon** [comment]
Managed to write the wrong version number,
`pip install --upgrade git+https://github.com/Egsagon/PHUB.git`
should work fine now. sorry -_-

**Comment by github-ph-52obr** [comment]
That did it! Thank you for your help and responsiveness!

**Comment by Egsagon** [comment]
No problem, have a great day!

**Comment by github-ph-52obr** [comment]
Hey, this is now causing the reverse issue where pornstars work but models don’t.

**Comment by Egsagon** [comment]
should work now.

**Comment by github-ph-52obr** [comment]
Models are working again. Pornstars are mostly working. It’s only picking up the first 18 videos for the few pornstars I tested.

**Comment by Egsagon** [comment]
Hi, pornstars now have their own object.

You can now choose if you want to iterate through the pornstar personal uploads or the professional videos they are featured in.

```python
ps = client.get_user('...')

for video in ps.uploads:
    ...
for video in ps.videos:
    ...
```

Hope this fixes it.
Keep in mind this object isn't fully implemented yet.

**Comment by github-ph-52obr** [comment]
I’ll give it a test once you push the update!

**Comment by github-ph-52obr** [comment]
I’m still on 4.1.4 but I noticed that:

```
user = client.get_user('this-user')
for video in user.videos:
    print(video.title)
```

now returns a bunch of random urls for other videos.

**Comment by Egsagon** [comment]
Hi, the update is already pushed with 9e7d951a9ceb9f67155e94bca54a07fa92bed402, i just forgot to change the version number (again). It also fixes the 4.1.4 random urls.

**Comment by github-ph-52obr** [comment]
Ok I've updated it to v4.1.4-2, but how do I differentiate between user and pornstar in the code to run? Is there a new argument in the .get_user() function? Sorry, still learning all this

**Comment by Egsagon** [comment]
Pornstars objects inherit from User objects. This mean they are pretty much the exact same. This is why, the Pornstar.videos property has the same name as User.videos so you don't have to make the difference between those objects.

If you still want to for some reason, you can either compare their type using python isinstance() or type() function, or use the User.type (or Pornstar.type) which is a string containing the type of the user (pornstar, model, channel, member, etc.).

But in most cases, you won't care about what is outputed from client.get_user():

```python
user = client.get_user('pornstar_or_model_url')

for video in user.videos[:10]:
   # Iterate through pornstar or model videos
```

**Comment by github-ph-52obr** [comment]
Coming back to this. Seems like it still struggles in certain instances. Take Alice Green as an example: https://www.pornhub.com/pornstar/alice-green

Her page doesn’t have a separate videos or uploads section.

This returns an empty list:
`user = client.get_user('https://www.pornhub.com/pornstar/alice-green')
urls = []
for video in user.videos:
    urls.append(video.url)
urls `

Whereas this returns an error:
`user = client.get_user('https://www.pornhub.com/pornstar/alice-green')
urls = []
for video in user.uploads:
    urls.append(video.url)
urls `

IndexError: list index out of range

**Comment by Egsagon** [comment]
Yes. Thanks for reporting. Queries are a mess right now, and i don't have the time to fix them.

**Comment by github-ph-52obr** [comment]
Hey, no worries! I know you’re dealing with personal issues. Just wanted to raise this. No rush!

**Comment by Egsagon** [comment]
Hey, this should be fixed with 89043a6a9494422a834f34171f44a8dfaacc4a5e. It's on a dev branch because i still have other things to fix before i make a release.

Sorry for the wait,
tell me if it works for you or you find any bug!
